secant/Utils/Zatoshi.swift

```diff
@@ -0,0 +1,103 @@
+//
+//  Zatoshi.swift
+//  secant-testnet
+//
+//  Created by Lukáš Korba on 26.05.2022.
+//
+
+import Foundation
+
+typealias ZEC = NSDecimalNumber
```

Comment:
Do we want to treat Zec as a "first class citizen" in terms of being a numeric value?

I've been thinking about it as basically just a UI string for display only, with the intention of preventing Zec from ever being used for anything other than display.

As a NSDecimalNumber, a client could theoretically add/subtract Zec values and go outside of the valid range for Zatoshi (when converted back).  As a string, a client would need to go to Zatoshi and do the operations there, then come back to the string representation.

Both ways could work, although always using Zatoshi seems to reduce our surface area for bugs.

Reply:
reimplemented, the ZEC is no more, we always work with Zatoshi and only when we want to display the value we "convert" it using .asHumanReadableZec (returning String)